ui/src/pages/MenuEditor/Preview.tsx

```
@@ -0,0 +1,118 @@
+    onMenuClick?: (key: string) => void;
+    navigate?: NavigateFunction;
+}
+
+
+const Preview: React.FC<PreviewProps> = ({ menuData, navigate: propNavigate }) => {
+    // 尝试使用传入的 navigate 或者 useNavigate hook
+    let navigateFunc: NavigateFunction | undefined;
+
+    try {
+        const routerNavigate = useNavigate();
+        navigateFunc = propNavigate || routerNavigate;
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)